### ISSUE TITLE
Remove PnnsGroupQueryConfiguration, use ProductSearchQueryConfiguration instead

### ISSUE THREAD

**Issue by stephanegigandet** [openfoodfacts/openfoodfacts-dart#92]
There shouldn't be a need for special classes and methods to make search queries for a specific PNNS group. Those queries can be supported by the normal search queries, by specifying a tag filter on pnns_groups_1 or pnns_groups_2

e.g. https://world.openfoodfacts.org/pnns-group-2/biscuits-and-cakes is the same as https://world.openfoodfacts.org/search?pnns_groups_2_tags=biscuits-and-cakes

I doubt that any app is using those PNNS queries (except for Smoothie), but it will be a breaking change, so we should update the version to from 0.3.something to 0.4.0 when we do that.

In the mean time, maybe we could display some kind of warning to indicate that those functions are deprecated.

something like this should work:

```````
      var parameters = <Parameter>[
        const OutputFormat(format: Format.JSON),
        const Page(page: 5),
        const PageSize(size: 10),
        const SearchSimple(active: true),
        const SortBy(option: SortOption.PRODUCT_NAME),
        const TagFilter(
            tagType: "pnns_groups_2",
            contains: true,
            tagName: "biscuits-and-cakes"),
      ];

      ProductSearchQueryConfiguration configuration =
          ProductSearchQueryConfiguration(
              parametersList: parameters,
              fields: [ProductField.ALL],
              language: OpenFoodFactsLanguage.FRENCH);

      SearchResult result = await OpenFoodAPIClient.searchProducts(
          TestConstants.TEST_USER, configuration,
          queryType: QueryType.TEST);

**Comment by peterwvj** [comment]
Sounds good. For what it's worth, I don't use this feature. We might even want to bump the major version since this is a breaking change.

**Comment by monsieurtanuki** [comment]
Probably related: when I see product `4337185662713`, I find the following data very interesting:
```json
"categories_tags":[
"en:plant-based-foods-and-beverages",
"en:plant-based-foods",
"en:cereals-and-potatoes",
"en:breakfasts",
"en:cereals-and-their-products",
"en:breakfast-cereals",
"en:cereals-with-fruits",
"en:mueslis",
"en:mueslis-with-fruits"
]
```
Is it possible to use those data for a search about similar products, and how should we write the filter? (cf. https://github.com/openfoodfacts/smooth-app/issues/23)

PS: I recommend the annotation `@deprecated`

**Comment by stephanegigandet** [comment]
@monsieurtanuki : the categories_tags correspond to our multilingual and multilevel categories taxonomy. When someone adds a category, we will also all the parents categories. But that taxonomy is very complex: it can be very detailed (very fine categories), some categories have multiple parents etc. So it's a bit difficult to navigate. e.g. try to start at "Snacks" and to go down: https://world.openfoodfacts.org/category/snacks

The PNNS groups 1 and 2 are a 2 level hierarchy (it is the groups used by the French "Programme National Nutrition et Santé"). It's built on top of the categories. @PrimaelQuemerais chose it for Smoothie because the PNNS groups are relatively balanced.

**Comment by monsieurtanuki** [comment]
@stephanegigandet Oh, I don't mean to dismiss PNNS groups and use only categories_tags instead.

The idea was to keep PNNS as is, fix the query as this issue is about, and to introduce another type of search.
No navigation, just: I see a product, I would like to see similar products, I see all the categories it belongs to in `categories_tags`, I select the category that match the desired granularity, I click, and then I see the similar products.

**Comment by stephanegigandet** [comment]
@monsieurtanuki Sorry for the late response, I didn't see your last comment. Yes, definitely, the categories_tags can be useful to find similar products. The last one is the most specific.

The query would be something like:

```````
var parameters = <Parameter>[
        const TagFilter(
            tagType: "categories",
            contains: true,
            tagName: "en:mueslis-with-fruits"),
      ];

**Comment by monsieurtanuki** [comment]
Thank you @stephanegigandet! Cf. https://github.com/openfoodfacts/smooth-app/pull/183

**Comment by monsieurtanuki** [comment]
Fixed by #181.
